lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1174,7 +1176,9 @@ def _handle_abuse_flagged_field(form_value, user, cc_content):
             else:
                 comment_flagged.send(sender='flag_abuse_for_comment', user=user, post=cc_content)
     else:
-        cc_content.unFlagAbuse(user, cc_content, removeAll=False)
+        remove_all = bool(user.id != cc_content["user_id"] and is_privileged_user(course_key,
```

Comment:
What is this first check for?

Reply:
It is to check if the user that requested the unflagging of content is not the author of the content.
We don't want moderators to unreport all abuse flaggers for their own posts!